clients/src/main/java/org/apache/kafka/common/requests/FetchRequest.java

```diff
@@ -58,16 +61,18 @@
 
     public static final class PartitionData {
         public final long offset;
+        public final long logStartOffset;
         public final int maxBytes;
 
-        public PartitionData(long offset, int maxBytes) {
+        public PartitionData(long offset, long logStartOffset, int maxBytes) {
             this.offset = offset;
+            this.logStartOffset = logStartOffset;
```

Comment:
I didn't change it in order to reduce the amount of code change and conflicts with other patches. I will change it as you suggested.